### PR TITLE
Sheet stacks change icon depending on how full they are.

### PR DESCRIPTION
**Pull request by Kurgis**
Places all remaining stack items into their own file, that haven't already.

Adds item_state change depending on how large a stack of material sheets are. Currently this only applies to plasteel, metal, and plastic.
zzzfeature
Uses the code from https://github.com/tgstation/tgstation/pull/27964

Adjusts the tile parent item to hold the vars, and you can no longer throw them three screens away.

Deletes unused icons in obj\mining.dmi

### CODE REVIEW

**Comment by Kurgis** [comment]
Please review the stack.dm changes closely. Tested in-game, but not sure if all update_icon() are correct.

**Comment by mistyLuminescence** [comment]
sexy as heck

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 1]
```diff
@@ -9,12 +9,15 @@
...
+	singular_name = "broken tile"
 	icon = 'icons/obj/stack/tile.dmi'
-	icon_state = "tile"
-	desc = "A non-descript floor tile"
+	desc = "This should not exist."
 	w_class = ITEM_SIZE_NORMAL
+	force = WEAPON_FORCE_HARMLESS
+	throwforce = WEAPON_FORCE_WEAK
+	throw_speed = 3
+	throw_range = 7
```
[the remark this answers is not in the document]
Comment (Kurgis) [reply code/game/objects/items/stacks/tiles/tile_types.dm]: The throw speed prevents the tile from suddenly jumping to where it was thrown, the throw range prevents using it as ghetto artillery, and the force changes are for grass/carpet tiles to save line space. It is also to cover the force of any tile that's not under `tile/floor`.

[the remark this answers is not in the document]
Comment (drexample) [reply code/game/objects/items/stacks/tiles/tile_types.dm]: it isn't quite the topic of the PR, but the change is not unwelcome, so alright.

**Comment by drexample** [comment]
Tested in-game, seems to work fine.

**Pull request merged**